lib/utils/class-alerts.php

```diff
@@ -0,0 +1,270 @@
+	}
+
+	/**
+	 * Send the alert
+	 *
+	 * @param array $body The alert message body
+	 *
+	 * @return array Response details from wp_remote_post
+	 */
+	protected function send( $body ) {
+		$response = wp_remote_post( $this->service_url, [
+			'timeout' => 0.1,
+			'body' => json_encode( $body ),
+		] );
+
+		if ( is_wp_error( $response ) ) {
+			throw new Exception( sprintf( 'Error sending alert: %s', $response->get_error_message() ) );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Cache alert
+	 *
+	 * Used to throttle alerts requests
+	 *
+	 * @param $kind string Cache key
+	 * @param $interval integer When the cache data should expire, in seconds
+	 *
+	 * @return bool
+	 */
+	protected function add_cache( $key, $expire ) {
+		if ( function_exists( 'wp_cache_add' ) && function_exists( 'wp_cache_add_global_groups' ) ) {
+			wp_cache_add_global_groups( [ 'irc-ratelimit' ] );
+
+			return wp_cache_add( $key, 1, 'irc-ratelimit', $expire );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validate IRC channel or user name
+	 *
+	 * @param $channel_or_user string
+	 *
+	 * @return string Validated and cleaned channel or user name
+	 */
+	protected function validate_channel_or_user( $channel_or_user ) {
+		$channel_or_user = preg_replace( '/[^0-9a-z#@|.-]/', '', $channel_or_user );
+
+		if ( ! $channel_or_user ) {
+			throw new Exception( "Invalid \$channel_or_user: Alerts\:\:chat( '$channel_or_user' );" );
+		}
+
+		return $channel_or_user;
+	}
+
+	/**
+	 * Validate alert message
+	 *
+	 * Ensure the message is a valid string
+	 *
+	 * @param $message mixed
+	 *
+	 * @return string Validated and trimmed message
+	 */
+	protected function validate_message( $message ) {
+		if ( is_array( $message ) || is_object( $message ) ) {
+			throw new Exception( "Invalid \$message: Alerts\:\:chat( " . print_r( $message, true ) . " );" );
+		}
+
+		$message = trim( $message );
+
+		if ( ! $message ) {
+			throw new Exception( "Invalid \$message: Alerts\:\:chat( '$message' );" );
+		}
+
+		return $message;
+	}
+
+	/**
+	 * Validate Opsgenie details array
+	 *
+	 * @param $details array
+	 *
+	 * @return array
+	 */
+	protected function validate_details( $details) {
+		$required_keys = [ 'alias', 'description', 'entity', 'priority', 'source' ];
+
+		if ( ! is_array( $details ) ) {
+			throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+		}
+
+		foreach( $details as $key => $value ) {
+			if ( ! array_key_exists( $key, $required_keys ) ) {
+				throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+			}
+
+			if ( ! $value ) {
+				throw new Exception( "Invalid \$details: Alerts\:\:opsgenie( " . print_r( $details, true ) ." );" );
+			}
+		}
+
+		return $details;
+	}
+
+	/**
+	 * Get an instance of this Alerts class
+	 *
+	 * @return Alerts
+	 */
+	public static function get_instance() {
```

Comment:
For singletons, the common convention is to name the function as `instance()`.
```suggestion
	public static function instance() {
```